src/viper/syntax.ml

```diff
@@ -41,6 +45,7 @@ and exp' =
   | AndE of exp * exp
   | OrE of exp * exp
   | Implies of exp * exp
+  | Old of exp
```

Comment:
Conceptually, is `old` just a flag on `VarE`? (I am not suggesting to change, just want to understand.)

Reply:
In simple cases yes, but one old expression can wrap multiple variables (it distributes over everything, more or less). 

For example `old(x + y)` is the same value as `old(x) + old(y)` (assuming that `x` and `y` are actor fields).

Reply:
In general, we could also have `old(foo())` with, e.g.: 

```motoko
private func foo() : Int {
    x
}
```

where `x` is a private field.